rfc/development-cycle.md

```diff
@@ -0,0 +1,180 @@
+(this also falls out of the above assumption). 
+
+* Patch: No commits towards `{x}.{y}.{z}` till `{x}.{y}.{z-1}` final is released.
+* Minor: No commits towards `{x}.{y}.0` till `{x}.{y-1}.0` final is released.
+* Major: No commits towards `{x}.0.0` till `{x-1}.0.0` final is released.
+
+The last two mean you have to totally finish release before you start on the next one. For example, it would be illegal to have releases in this order: `1.2.0a0`, `1.3.0a0`, `1.2.0`, then `1.3.0`.
+
+**JavaScript versions In Sync** We are ok always keeping the JS version bumps in sync. Meaning that if we do a major release of the Python package we also do a major release of all JS versions.
+
+## Strategy
+
+Based on these assumption we can create a consistant branching strategy.
+
+### Branches
+
+Each branch should be aligned to different releases:
+
+1. Patch: `{x}.{y}.x` branches.
+2. Minor: `{x}.x.0` branches. After final release `{x}.{y}.0` create new patch `{x}.{y}.x` branch.
+3. Major: `master` branch. After release `{x}.0.0` create new minor `{x}.x.0` branch and new patch `{x}.0.x` branch.
+
+Each increment has a corresponding "latest" branch for the greatest working version. For example with two branches `2.x.0`, `1.x.0`  the "latest" minor branch is `2.x.0`. The latest major branch is always `master`.
+
+*Note: We could instead have separate branches for each version, but this isn't necessarily if we are going by the **Single Active** assumption above.*
+
+
+### PR Semver Tags
+
+Each PR should be tagged with one of three tags `semver:patch`, `semver:minor`, `semver:major` before it is merged. We could do this manually or by possibly giving a first guess [using type analysis](https://api-extractor.com/) or some commit message keywords. We should have a bot that blocks merging if one of these is not added.
+
+If you have change that is *only* a backprt, and should not appear in master, then you can target that branch as the base of this merge.
+
+### PR Backports
+Once a PR is merged into master, backport PRs should be opened against all other open branches of the corresponding increment. 
+This will likely result in some cases where backports are proposed that are not appropriate, in which case they can be closed. 
+
+1. Patch: Backport to all `{x}.{y}.x` branches and all `{x}.x.0` branches.
+2. Minor: Backport to all `{x}.x.0` branches.
+3. Major: No backports.
+
+
+### Milestones
+
+Each branch therefore always has an "active" milestone associated with it, which corresponds to the next final release
+on that branch. If we don't wish to make any more releases from a branch, we should delete it.
+
+Each PR's milestone should reflect active milestone of the branch it targets.
+
+### Changelog
+
+Each PR should add or edit a file in a directory full of files corresponding to unreleased changes, [like matplotlib does](https://matplotlib.org/devel/contributing.html#contributing-pull-requests). During a final release, these items are deleted
```

Comment:
Love it.  We should also have the bot block when there is no change to this file